src/condor_utils/file_transfer.cpp

```diff
@@ -3097,7 +3097,7 @@ FileTransfer::DoDownload( filesize_t *total_bytes_ptr, ReliSock *s)
 					errstack.getFullText().c_str() );
 				download_success = false;
 				hold_code = CONDOR_HOLD_CODE_DownloadFileError;
-				hold_subcode = rc;
+				hold_subcode = 0;
```

Comment:
This appears to be unrelated to correctly reporting the filename.  It should not be changed until we've decided what, if anything, we want `HoldReasonSubcode` to be when reporting plug-in failures; see HTCONDOR-842.

Reply:
There are no code paths where `rc` is nonzero here and fixes a compilation error when you declare `rc` inside the scope where it is expected to be used.

Determining `rc` must always be `0` here is really hard from just reading the code so this  is a significant piece of knowledge for the next person having to read the code.

Reply:
I have yet to be convinced that `0` is the _correct_ value for `hold_subcode`.  I'd rather mark it as a problem to be solved in another ticket than just change it, especially without explanation.

Reply:
I agree with Todd that `0` is not the correct value for `hold_subcode`, but since `rc` isn't the correct value either I think this is fine for now. We have another ticket open in [HTCONDOR-842](https://opensciencegrid.atlassian.net/browse/HTCONDOR-842) which deals with file transfer plugin error conditions, that seems like a better place to standardize our exit codes and put a meaningful value here.